server.py

```diff
@@ -0,0 +1,122 @@
+        self.server.listen()
+        print("Server started")
+        while True:
+            client, address = self.server.accept()
+            logger.info(f"{address} connected")
+            thread = threading.Thread(target=self.handle_client, args=(client, address))
+            thread.start()
+            print(f"Active connections: {threading.activeCount() - 1}")
+
+    def handle_client(self, client, address):
+        print(f"{address} connected")
+        client.send(self.WELCOME_MSG.encode(self.FORMAT))
+        while True:
+            msg = client.recv(2048).decode(self.FORMAT)
+            if msg == "/help":
+                client.send(self.HELP_MSG.encode(self.FORMAT))
+            elif msg == "/register":
+                self.register(client, self.db_path)
+            elif msg == "/login":
+                self.login(client, self.db_path)
+            elif msg == "/exit":
+                client.close()
+                print(f"{address} disconnected")
+                break
+            elif msg.startswith("@"):
+                self.send_to_client(msg)
+            else:
+                try:
+                    print(f"[{datetime.now().strftime(self.DATE_FORM)}]"
+                          f" {self.usernames_connected[self.clients_connected.index(client)]}:-> {msg}")
+                    self.broadcast(f"[{datetime.now().strftime(self.DATE_FORM)}]"
+                                   f" {self.usernames_connected[self.clients_connected.index(client)]}:->"
+                                   f" {msg}".encode(self.FORMAT))
+                    logger.info(f"{self.usernames_connected[self.clients_connected.index(client)]}"
+                                f" sent a broadcast message")
+                except:
+                    client.send("Only registered users can post. If you are registered try /login".encode(self.FORMAT))
+
+    def register(self, client, db_path):
+        db = Database(db_path)
+        client.send('Enter your username: '.encode(self.FORMAT))
+        username = client.recv(1024).decode(self.FORMAT)
+        while username in db.get_users_list("users"):
+            client.send("Username is already in use. Try again or /exit".encode(self.FORMAT))
+            username = client.recv(1024).decode(self.FORMAT)
+        client.send('Enter your password: '.encode(self.FORMAT))
+        password = client.recv(1024).decode(self.FORMAT)
+        while len(password) < 6:
+            client.send("Password too short. It should be at lest 5 symbols".encode(self.FORMAT))
+            password = client.recv(1024).decode(self.FORMAT)
+        db.insert_item("users", (username, password))
+        logger.info(f"New user registered. Username is {username}")
+        print(f"New user registered. Username is {username}")
+        client.send(f"You successfully registered as {username}. Now /login.".encode(self.FORMAT))
+
+    def login(self, client, db_path):
+        db = Database(db_path)
```

Comment:
why db is not set as instance attribute?
you will create instance of db every time when you need it? Or it is better to create it once and use

Reply:
Fixed